pyproject.toml

```diff
@@ -41,7 +41,7 @@ matplotlib = "^3.7.1"
 
 # Extras (keep versions in sync with dev deps)
 apache-beam = { version = "^2.48.0", optional = true }
-tensorflow = { version = "^2.12.0", optional = true }
+tensorflow = {version = ">=2.12.0,<3.0.0", optional = true}
```

Comment:
tensorflow 2.16 drops support for Keras 2 which might cause some issues.
It might be interesting to keep <2.16 for now.